app/models/facility.rb

```diff
@@ -79,8 +79,18 @@ class Facility < ApplicationRecord
   delegate :protocol, to: :facility_group, allow_nil: true
   delegate :organization, to: :facility_group, allow_nil: true
   delegate :follow_ups_by_period, to: :patients, prefix: :patient
-  delegate :diabetes_follow_ups_by_period, to: :patients
-  delegate :hypertension_follow_ups_by_period, to: :patients
+
+  def hypertension_follow_ups_by_period(*args)
+    patients
+      .hypertension_follow_ups_by_period(*args)
+      .where(blood_pressures: {facility: self})
+  end
```

Comment:
Iiinteresting, this wasn't capturing the facility scoping we do in [other parts of the dashboard](https://github.com/simpledotorg/simple-server/blob/progress-control-01/app/queries/district_analytics_query.rb#L47-L50), I see.

Reply:
Yep, this query in district analytics works fine because it's called through patient and the facility scoping happens later (even though with a different intention).